src/main/java/org/codehaus/plexus/components/io/resources/PlexusIoFileResource.java

```diff
@@ -94,14 +95,7 @@ private static DeferredFileOutputStream asDeferredStream( @Nonnull ContentSuppli
 
     private static ContentSupplier getRootContentSupplier( final File file )
     {
-        return new ContentSupplier()
-        {
-            public InputStream getContents()
-                throws IOException
-            {
-                return new FileInputStream( file );
-            }
-        };
+        return () -> Files.newInputStream( file.toPath() );
```

Comment:
Out of curiosity why `Files.newInputStream` instead of `new FileInputStream`?

Reply:
The FileInputStream and FileOutputStream classes contains a finalizer method which will cause garbage collection pauses. This issue was "fixed" in Java 10, yet since this project could still use Java 8 it's more than ideal to use a potentially better performant alternative.

A more complete explanation is here:
* https://pmd.github.io/latest/pmd_rules_java_performance.html#avoidfilestream